File: tech_yourself_ios7_in_21hours/HelloXCode/HelloXCode/ViewController.m
//
//  ViewController.m
//  HelloXCode
//
//  Created by mcxiaoke on 15/8/17.
//  Copyright (c) 2015年 mcxiaoke. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UILabel *label1;
@property (weak, nonatomic) IBOutlet UILabel *label2;
@property (weak, nonatomic) IBOutlet UIButton *button3;

@property(weak, nonatomic) IBOutlet UILabel* centerLabel;
@end

@implementation ViewController
- (IBAction)button1:(UIButton *)sender {
}
- (IBAction)button2:(UIButton *)sender {
}

- (void)viewDidLoad {
  [super viewDidLoad];
  CGFloat x = self.centerLabel.frame.origin.x;
  CGFloat y =
      self.centerLabel.frame.origin.y + self.centerLabel.frame.size.height + 10;
  NSLog(@"%g-%g", self.centerLabel.frame.origin.x,
        self.centerLabel.frame.origin.y);
  UILabel* aLabel = [[UILabel alloc] initWithFrame:CGRectMake(x, y, 200, 40)];
  aLabel.textAlignment = NSTextAlignmentCenter;
  aLabel.backgroundColor = [UIColor blueColor];
  aLabel.textColor = [UIColor whiteColor];
  aLabel.text = @"a custom label";
  [self.view addSubview:aLabel];

  NSLog(@"%g-%g", aLabel.frame.origin.x, aLabel.frame.origin.y);
}

#pragma mark <label>

- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}

@end

File: ios_programming_4th/Hypnosister-ch5/Hypnosister/ViewController.swift
//
//  ViewController.swift
//  Hypnosister
//
//  Created by Xiaoke Zhang on 2017/8/15.
//  Copyright © 2017年 Xiaoke Zhang. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
//        let firstFrame = CGRect(x:160, y:240, width: 100, height: 150)
        let firstFrame = self.view.bounds
        let sv1 = HypnosisterView(frame:firstFrame)
//        sv1.backgroundColor = UIColor.red
        self.view.addSubview(sv1)
        
        let secondFrame = CGRect(x:20, y:30, width:50, height:50)
        let sv2  = HypnosisterView(frame:secondFrame)
        sv2.backgroundColor = UIColor.blue
//        sv1.addSubview(sv2)
//        self.view.addSubview(sv2)
        
        self.view.backgroundColor = UIColor.white
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

